Added keyboard shortcuts for ghc-mod in haskell.

Tritlo opened the pull request:
I added some keyboard shortcuts for ghc-mod, to use in haskell-mode, since there were none previously. These include shortcuts to get info, type, and (the one which prompted this).

d12frosted commented:
Thank you for your PR. Indeed Spacemacs lacks better coverage for `ghc-mod` functions.

The only thing I would like to notice is that bindings for get info and get type functions are set [here](https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/haskell/packages.el#L135). This is done in such way due the fact that different 'extensions' to `haskell-mode` set this binding to the appropriate function, and we don't want to track which function should be used. Also Spacemacs has a convention about bindings for getting type of symbol - it should be `m h t`. So your commit kinda breaks it.

joehillen commented:
Should probably also add these bindings to the README.

robbyoconnor commented:
If you add keybindings, **ALWAYS, ALWAY, ALWAYS** document them!

d12frosted commented:
> Should probably also add these bindings to the README.

Yes, good call.

Tritlo commented:
Alright, I've fixed your grievances, and squashed the fixes together. I removed some redundancies and added documentation. Will you please review again?

d12frosted commented on layers/+lang/haskell/packages.el:
```diff
@@ -45,10 +45,25 @@
-      (set-face-attribute 'ghc-face-warn nil :underline nil))))
+    (progn
+      (if (fboundp 'which-key-declare-prefixes)
+          (which-key-declare-prefixes
```
Better to use `(spacemacs/declare-prefix-for-mode 'haskell-mode "mm" "haskell/ghc-mod")`.

d12frosted commented:
@Tritlo Thanks for the update! Sorry that I haven't noticed that thing with `prefix` before.

Tritlo commented:
Alright, added the prefix, and updated the documentation a bit to be more in line with the other documentation.

d12frosted commented:
Cool, @TheBB  summoning you :smile:

TheBB commented:
Thank you! Cherry-picked in develop. You can safely delete your branch.
